designsafe/apps/data/management/commands/es_setup.py

```diff
@@ -29,58 +29,53 @@ def add_arguments(self, parser):
+                return False
+            return response
 
+        indexes = ["designsafe-{}-files",
+                   "designsafe-{}-publications",
+                   "designsafe-{}-publications-legacy",
+                   "designsafe-{}-projects",
+                   "designsafe-{}-rapid-events",
+                   "designsafe-{}-rapid-event-types"
+                   ]
 
-        elasticsearch.helpers.reindex(
-            client=remote_es_client,
-            target_client=local_es_client,
-            source_index="designsafe-{}-files".format(remote),
-            target_index="designsafe-{}-files".format(local),
-            query={"query": {"prefix": {"path._exact": "/{}".format(username)}}}
-        )
-
-        elasticsearch.helpers.reindex(
-            client=remote_es_client,
-            target_client=local_es_client,
-            source_index="designsafe-{}-publications".format(remote),
-            target_index="designsafe-{}-publications".format(local)
-        )
-
-        elasticsearch.helpers.reindex(
-            client=remote_es_client,
-            target_client=local_es_client,
-            source_index="designsafe-{}-publications-legacy".format(remote),
-            target_index="designsafe-{}-publications-legacy".format(local)
-        )
-
-        elasticsearch.helpers.reindex(
-            client=remote_es_client,
-            target_client=local_es_client,
-            source_index="designsafe-{}-projects".format(remote),
-            target_index="designsafe-{}-projects".format(local)
-        )
-
-        elasticsearch.helpers.reindex(
-            client=remote_es_client,
-            target_client=local_es_client,
-            source_index="designsafe-{}-rapid-events".format(remote),
-            target_index="designsafe-{}-rapid-events".format(local)
-        )
-
-        elasticsearch.helpers.reindex(
-            client=remote_es_client,
-            target_client=local_es_client,
-            source_index="designsafe-{}-rapid-event-types".format(remote),
-            target_index="designsafe-{}-rapid-event-types".format(local)
-        )
-
-        logger.debug(remote_es_client.info())
+        failed_indexes = []
+        for index in indexes:
+            query = {"query": {"prefix": {"path._exact": "/{}".format(username)}}} if "files" in index else None
+            result = reindex(
+                source_index=index.format(remote),
+                target_index=index.format(local),
+                query=query
+            )
+            if not result:
+                failed_indexes.append(index.format(local))
+        logger.info("Finished indexing.")
+        if failed_indexes:
+            logger.error(f"Successfully reindex {len(indexes) - len(failed_indexes)}"
+                         f" of {len(indexes)} indexes\n"
+                         f"failed indexes are: {', '.join(failed_indexes)}\n"
+                         f"(check logs above for errors related to failures)")
```

Comment:
Feel free to push back on this, but since the indexed entities are being printed, it is impossible to scroll up and view the errors. Maybe it would be a good idea to add an option to run without the indexed entities being printed, or to collect the errors in a list like you did with the failed indices?

Reply:
> the indexed entities are being printed, it is impossible to scroll up and view the errors.

the indexed entities aren't being printed; it is just the `designsafe-{}-publications` that fails and the ES library prints out the whole document. but you can pipe to a file to see the error.

Reply:
Yeah, makes sense to me